File: python/insertion_sort.py

num_list = [3, 6, 5, 9, 1, 4, 2, 8, 7]

for i in range(1, len(num_list)):
    for j in range(i, 0, -1):
        if num_list[j-1] > num_list[j]:
            temp = num_list[j-1]
            num_list[j-1] = num_list[j]
            num_list[j] = temp
        else:
            break

print(num_list)

File: javascript/insertionSort.js
const numList = [3, 6, 5, 9, 1, 4, 2, 8, 7]

let temp = 0
for (let i=1 ; i<numList.length ; i++){
  for (let j=i ; j>0 ; j--){
    if(numList[j-1] > numList[j]){
      temp = numList[j-1]
      numList[j-1] = numList[j]
      numList[j] = temp
    }
    else{
      break
    }
  }
}

console.log(numList)